<launch>
    <!-- ORB-SLAM3 -->
    <node name="orb_slam3_stereo_inertial" pkg="orb_slam3_ros_wrapper" type="orb_slam3_ros_wrapper_stereo_inertial" output="screen">
        <!-- for EuRoC dataset -->
        <remap from="/camera/left/image_raw"    to="/cam0/image_raw"/>
        <remap from="/camera/right/image_raw"   to="/cam1/image_raw"/>
        <remap from="/imu"                      to="/imu0"/>

        <!-- Parameters for original ORB-SLAM3 -->
        <param name="voc_file"      type="string"   value="$(find orb_slam3_ros_wrapper)/config/ORBvoc.txt" />
        <param name="settings_file" type="string"   value="$(find orb_slam3_ros_wrapper)/config/EuRoC_stereo_inertial_non_rectified.yaml" />
        <param name="do_rectify"    type="bool"     value="true" />
        <param name="do_equalize"   type="bool"     value="true" />

        <!-- Parameters for ROS -->
        <param name="map_frame_id"  type="string"   value="world" />
        <param name="pose_frame_id" type="string"   value="camera" />
    </node>

    <!-- Visualization - RViz-->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find orb_slam3_ros_wrapper)/config/orb_slam3.rviz" output="screen" />

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="orb_slam3_ros" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/camera" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>
</launch>